core/src/main/java/com/yandex/yoctodb/util/immutable/impl/FixedLengthByteArrayIndexedList.java

```diff
@@ -60,6 +62,42 @@ public Buffer get(final int i) {
         return elements.slice(((long) i) * elementSize, elementSize);
     }
 
+    @Override
+    public long getLongUnsafe(final int i) {
+        assert 0 <= i && i < elementCount;
+        assert elementSize == Long.BYTES;
+
+        return elements.getLong() ^ Long.MIN_VALUE;
+    }
+
+    @Override
+    public int getIntUnsafe(final int i) {
+        assert 0 <= i && i < elementCount;
+        assert elementSize == Integer.BYTES;
+
+        return elements.getInt() ^ Integer.MIN_VALUE;
+    }
+
+    @Override
+    public short getShortUnsafe(final int i) {
+        assert 0 <= i && i < elementCount;
+        assert elementSize == Short.BYTES;
+
+        final int res = elements.getShort() ^ Short.MIN_VALUE;
+
+        return Shorts.checkedCast(res);
+    }
+
+    @Override
+    public char getCharUnsafe(final int i) {
+        assert 0 <= i && i < elementCount;
+        assert elementSize == Character.BYTES;
+
+        final int res = elements.getChar() ^ Character.MIN_VALUE;
```

Comment:
`^ Character.MIN_VALUE` is redundant because it's XOR with zero, which doesn't change anything.

Reply:
Agree, fixed.